src/boss/mar/AFC4.c

```diff
@@ -21,4 +21,50 @@ INCLUDE_ASM("boss/mar/nonmatchings/AFC4", func_us_8018B74C);
 
 INCLUDE_ASM("boss/mar/nonmatchings/AFC4", EntityMariaCutscene);
 
-INCLUDE_ASM("boss/mar/nonmatchings/AFC4", func_us_8018C90C);
+extern s32 D_80097400[];
+extern u16 D_80180A60;
+extern u16 D_us_8019AF2A[];
+
+void func_us_8018C90C(Entity* self) {
+    Entity* player;
+
+    player = &PLAYER;
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(&D_80180A60);
+        D_8003C8B8 = 0;
+        D_80097400[0] = 1;
+        g_Player.padSim = PAD_RIGHT;
+        g_Player.D_80072EFC = 1;
+        return;
+    case 1:
+        if ((player->posX.i.hi >= 0x21) && !(g_CutsceneFlags & 1)) {
+            g_CutsceneFlags |= 1;
+        }
+        if (player->posX.i.hi >= 0xB1) {
+            g_Player.padSim = PAD_LEFT;
+            self->step += 1;
```

Comment:
These are usually an increment operator (`++`). It _usually_ doesn't matter for PSX, but almost always matters for PSP.